src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -231,6 +231,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
   let isCommitting: boolean = false;
   let isUnmounting: boolean = false;
 
+  // Use these to prevent an infinite loop of nested updates
+  let didCommit = false;
+  let nestedSyncUpdates = 0;
```

Comment:
Oops. Fixed to reset at the end of `performWork`.

Before fixing, I changed the limit to 3 and ran the tests again, got lots of failures as expected. So I guess in our whole test suite we have fewer than 1000 nested updates.